src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -67,6 +67,15 @@ public virtual void AddCompilationRoots()
 
                 AddCompilationRootsForExports(module);
             }
+
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
+            {
+                if (_typeSystemContext.InputFilePaths.ContainsKey(inputFile.Value)) continue;
+
+                var module = _typeSystemContext.GetModuleFromPath(inputFile.Value);
+                if (module.Equals(_typeSystemContext.SystemModule)) continue;
```

Comment:
I think this is only needed because there is

```csharp
AddCompilationRootsForExports((EcmaModule)_typeSystemContext.SystemModule);
```

In `SingleFileCompilationModuleGroup.cs`. The right fix is to delete the line from there.